### PR TITLE
ch6006 - fix and improve generation and level functions

### PR DESCRIPTION
**Pull request by DrDonk**

### CODE REVIEW

File: plaidcloud/utilities/remote/dimension.py [hunk 1]
```diff
@@ -838,13 +876,27 @@ def get_leaves(self, node, hierarchy=MAIN):
...
+            hierarchy (str): Hierarchy unique ID
+
+        Returns:
+            list: List of leaf level node objects
+        """
+        return self.dim.get_leaves_at_generation(project_id=self.project_id, name=self.name, node=node, generation=generation, hierarchy=hierarchy)
+
     def get_leaves_at_level(self, node, level, hierarchy=MAIN):
-        """get_leaves_at_level(node, level, hierarchy=MAIN)
+        """get_leaves_at_generation(node, level, hierarchy=MAIN)
```
Comment (rad-pat) [inline plaidcloud/utilities/remote/dimension.py:894]: wrong docu

File: plaidcloud/utilities/remote/dimension.py [hunk 2]
```diff
@@ -638,19 +661,33 @@ def shift_node_down(self, parent, child, hierarchy=MAIN):
     # --------------------------------------------------------------------------------------------------
     # ==== NAVIGATION METHODS ==========================================================================
     # --------------------------------------------------------------------------------------------------
-    def get_ancestor(self, node, level, hierarchy=MAIN):
-        """get_ancestor(node, level, hierarchy=MAIN)
+    def get_ancestor_at_generation(self, node, generation, hierarchy=MAIN):
+        """get_ancestor_at_level(node, generation, hierarchy=MAIN)
```
Comment (rad-pat) [inline plaidcloud/utilities/remote/dimension.py:665]: wrong docu

**Review by rad-pat — CHANGES_REQUESTED** [review]

File: plaidcloud/utilities/remote/dimension.py (hunk 1 above)
Comment (DrDonk) [reply plaidcloud/utilities/remote/dimension.py]: Fixed

File: plaidcloud/utilities/remote/dimension.py (hunk 2 above)
Comment (DrDonk) [reply plaidcloud/utilities/remote/dimension.py]: Fixed

**Review by rad-pat — APPROVED** [review]

**Pull request merged**